src/pip/_internal/cli/base_command.py

```diff
@@ -199,6 +199,20 @@ def _main(self, args):
                 )
                 options.cache_dir = None
 
+        if getattr(options, "build_dir", None):
+            deprecated(
+                reason=(
+                    "The -b/--build/--build-dir/--build-directory "
+                    "option is deprecated and has no effect anymore."
+                ),
+                replacement=(
+                    "use the TMPDIR/TEMP/TMP environment variable, "
+                    "possibly combined with --no-clean"
+                ),
+                gone_in="21.1",
```

Comment:
Should this be 21.0 instead? I don’t imagine PyCharm would need more than three months to release an upgrade.

Reply:
They requested 6 months in the issue.

Reply:
@uranusjr We've already released upgrades of the IDE, but we'd like to have at least one extra release cycle before the option is removed so that most of them have updated naturally, not because something had broken down. Sorry for causing all this hassle, and postponing a long-expected refactoring in your codebase.

Reply:
Thanks for the clarification!